### ISSUE TITLE
Training loss doesn't decrease on TPU while works fine on GPU

### ISSUE THREAD

**Issue by AbuUbaida** [pytorch/xla#3675]
## ❓ Questions and Help
For the summarization task, where I train with an encoder-decoder model on GPU, it works fine and the loss gets lower over iterations. But when I change the device to `device = xm.xla_device()` and optimizer to `xm.optimizer_step(optimizer, barrier=True)` on single-core TPU, the training loss remains nearly unchanged!!

**Here's the reproducible code:** https://colab.research.google.com/drive/1pC2CF3ipWt0eJrfXdznwAZD3zs0sX1kd?usp=sharing
Is it a bug or I am missing something?

**Comment by AlexWertheim** [comment]
Thanks for sharing this issue. In your training loop, can you try replacing `model.train()` with `model.train().to(device)`? Please also note that in your code, `xm.optimizer_step(optimizer, barrier=True)` is currently commented out in favor of `optimizer.step()`.

After I made these modifications, your code seems to work fine: loss steadily decreases with each epoch. (In my specific case, loss started at ~11.599, and ended at ~1.011 after the final epoch.)

**Comment by AbuUbaida** [comment]
@AlexWertheim Thanks a lot for your support! I am getting weak to the open source community.
